### PR TITLE
Add files with metadata info to the Photo Library.

### LINKED ISSUE #226: Photos save used the front camera where not being saved to the Selfies album
### Expected behavior

If inside the Selfies album, tap to capture a photos and take a photo with the front camera, the photo should be saved to selfies album

### Actual behavior

- Photo is saved and selected
- But doesn't show on the selfie album.

### Steps to reproduce the behavior

- Start the demo app
- Go to full screen picker mode
- Select Selfie album
- Tap on camera preview/capture
- Take a photo using the front camera
- Accept the photo
- Photo doesn't show on selfie album

### PR DESCRIPTION
**Pull request by SergioEstevao**
Fixes #226 

To test:
 - see ticket description above.

### CODE REVIEW

File: Pod/Classes/WPImageExporter.m [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    NSAssert(fileExtension.length > 0, @"file Extension cannot be empty");
+    NSString *fileName = [NSString stringWithFormat:@"%@_file.%@", NSProcessInfo.processInfo.globallyUniqueString, fileExtension];
+    NSURL * fileURL = [[NSURL fileURLWithPath: NSTemporaryDirectory()] URLByAppendingPathComponent:fileName];
+    return fileURL;
+}
+
++ (BOOL)writeImage:(UIImage *)image metadata:(NSDictionary *)metadata toURL:(NSURL *)url 
+{
+    NSMutableDictionary *properties = [[NSMutableDictionary alloc] initWithDictionary:@{ (NSString *)kCGImageDestinationLossyCompressionQuality: @(1)}];
+    CGImageDestinationRef destination = CGImageDestinationCreateWithURL((CFURLRef)url, kUTTypePNG, 1, nil);
```
Comment (frosty) [inline Pod/Classes/WPImageExporter.m]: Do we not want JPG here?

File: Pod/Classes/WPImageExporter.h [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#import <Foundation/Foundation.h>
+
+@interface WPImageExporter : NSObject
```
Comment (frosty) [inline Pod/Classes/WPImageExporter.h]: Could you write a quick doc comment explaining what this class is for / why we need it?

File: Pod/Classes/WPImageExporter.h [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
++ (BOOL)writeImage:(UIImage *)image metadata:(NSDictionary *)metadata toURL:(NSURL *)url;
+
++ (NSURL *)URLForTemporaryFileWithFileExtension:(NSString *)fileExtension;
```
Comment (frosty) [inline Pod/Classes/WPImageExporter.h]: Nitpick, but we could probably shorten this a tiny bit: `temporaryFileURLWithExtension:`

**Comment by SergioEstevao** [comment]
@frosty ready for another look.

File: Pod/Classes/WPImageExporter.h [hunk 4]
```diff
@@ -1,9 +1,25 @@
...
+ @return an URL for a temporary file.
+ */
++ (NSURL *)temporaryFileURLWithExtension:(NSString *)fileExtension;
+
+/**
+ Writes an UIImage with the provided metadata to the designated URL using the JPG format.
+
+ @param image the image to save
+ @param metadata the metadata of the image
+ @return the URL of the image if it was save properly.
```
Comment (frosty) [inline Pod/Classes/WPImageExporter.h]: Typo: "save" -> "saved"

File: Pod/Classes/WPImageExporter.m [hunk 5]
```diff
@@ -6,18 +6,24 @@
...
 @implementation WPImageExporter
 
-+ (NSURL *)URLForTemporaryFileWithFileExtension:(NSString *)fileExtension
+/**
+ Retrieve an URL for a file on the temporary folder using the extension provided
+
+ @param fileExtension the extension to use.
+ @return an URL for a temporary file.
+ */
++ (NSURL *)temporaryFileURLWithExtension:(NSString *)fileExtension
```
Comment (frosty) [inline Pod/Classes/WPImageExporter.m]: Don't need comments here, as we have them in the header?

**Pull request merged**